File: server/feed14/service/Models/Participant.cs
namespace newsfeed.Models;

public class Participant
{
    public Participant(string id, string name) {
        Id = id;
        Name = name;
        Link = $"/participant/{id}";
    }
    
    public string Id { get; }

    public string Name { get; }

    public string Link { get; }
}

File: server/feed8/src/main/java/info/glennengstrand/api/Participant.java
package info.glennengstrand.api;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.annotation.JsonAutoDetect;
import org.springframework.validation.annotation.Validated;

/**
 * Participant
 */
@Validated
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record Participant(
    @JsonProperty("id") Long id,
    @JsonProperty("name") String name,
    @JsonProperty("link") String link
) {
}
